src/pkg/artifact/manager_proxy.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifact
+
+import (
+	"context"
+	"time"
+
+	"github.com/goharbor/harbor/src/lib/cache"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/retry"
+)
+
+var _ Manager = &managerProxy{}
+
+// managerProxy proxy the manager operations.
+type managerProxy struct {
+	*manager
+	// cache returns cache client.
+	cache func() cache.Cache
+	// keyBuilder builds cache object key.
+	keyBuilder *cache.ObjectKey
+	// lifetime is the cache life time.
+	lifetime time.Duration
+}
+
+// NewManagerProxy returns an instance of the manager proxy.
+func NewManagerProxy(m *manager, c func() cache.Cache, kb *cache.ObjectKey, lifetime time.Duration) Manager {
+	return &managerProxy{
+		manager:    m,
+		cache:      c,
+		keyBuilder: kb,
+		lifetime:   lifetime,
+	}
+}
+
+func (m *managerProxy) Get(ctx context.Context, id int64) (*Artifact, error) {
+	key, err := m.keyBuilder.Format("id", id)
+	if err != nil {
+		return nil, err
+	}
+
+	art := &Artifact{}
+	if err = m.cache().Fetch(ctx, key, art); err == nil {
```

Comment:
Suggest to use FetchOrSave
https://github.com/goharbor/harbor/blob/5be0b93caaf88cd9a018d678cca5885edccc32b4/src/pkg/config/db/cache.go#L45

Reply:
`FetchOrSave` mix the error for `query from db` and `save to cache`, so we can't distinguish the type of error, but this case is important for cache layer because we can ignore the error of save to cache, next GET will retry this operation.